source/extensions/tracers/xray/config.cc

```diff
@@ -0,0 +1,46 @@
+#include "extensions/tracers/xray/config.h"
+
+#include <string>
+
+#include "envoy/registry/registry.h"
+
+#include "common/common/utility.h"
+#include "common/tracing/http_tracer_impl.h"
+
+#include "extensions/tracers/well_known_names.h"
+#include "extensions/tracers/xray/xray_tracer_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace XRay {
+
+XRayTracerFactory::XRayTracerFactory() : FactoryBase(TracerNames::get().XRay) {}
+
+Tracing::HttpTracerPtr
+XRayTracerFactory::createHttpTracerTyped(const envoy::config::trace::v2::XRayConfig& proto_config,
+                                         Server::Instance& server) {
+  std::string sampling_rule_json;
+  try {
+    sampling_rules_json = server.api().fileSystem().fileReadToEnd(proto_config.json_file());
+  } catch (EnvoyException& e) {
+    ENVOY_LOG(error, "Could not read custom sampling rule json file: {}, because of {}.",
+              proto_config.json_file(), e.what());
+  }
```

Comment:
What happens if there are no rules?  Does the filter handle the case where `sampling_rules_json` is empty?

From what I see later, this ends up in `sampling_rules_` which isn't used.   Did I miss something along the way?

Reply:
if there are no rules, then `sampling_rules_` will be empty.
Since a skeleton only, no sampling logic is there yet. But, I think if we fail to load any rules, from the JSON file, then we still have the rules that we can load from the xray service.
If both can't be loaded, then we sample nothing. It acts like a NullTracer

Reply:
OK, makes sense 👍